kratos/processes/find_global_nodal_elemental_neighbours_process.cpp

```diff
@@ -30,12 +31,7 @@ namespace Kratos
 
         //first of all the neighbour nodes and elements array are initialized to the guessed size
         //and empties the old entries
-        #pragma omp parallel for
-        for(int i=0; i<static_cast<int>(rNodes.size()); ++i)
-        {
-            auto in = rNodes.begin() + i;
-            in->SetValue(NEIGHBOUR_ELEMENTS, GlobalPointersVector< Element >());
-        }
+        VariableUtils().SetVariable(NEIGHBOUR_ELEMENTS, GlobalPointersVector< Element >(), rNodes);
```

Comment:
```suggestion
        VariableUtils().SetNonHistoricalVariable(NEIGHBOUR_ELEMENTS, GlobalPointersVector< Element >(), rNodes);
```

this should fix the CI failure
Didn't you get this locally

(please check the function name I am not sure if this is the one)

Reply:
@ashishdarekar is probably makes sense for you to read this: https://github.com/KratosMultiphysics/Kratos/wiki/How-to-Access-DataBase

Reply:
Thanks for the reference. I have read it . I understood the major difference now. 
I was using SetVariable() (which is for Historical Type of data) ,Here **NEIGHBOUR_ELEMENTS** is nonHistorical Variable. So to use SetNonHistoricalVariable() make sense.

Reply:
> this should fix the CI failure
> Didn't you get this locally
> 
> (please check the function name I am not sure if this is the one)

Sorry, but I didnt run any of the tests.I have just compiled it, which didnt show any error. 
But I should have run the test before only locally. 
By the way I have a doubt, which test I should have done, for this particular mistake to detect?

Reply:
To be safe you can do the same as the CI
=> `runkratos kratos/python_scripts/run_tests.py`

This runs the python and c++ tests of the Core and all the Apps you have compiled and is generally the safest choice. It can however be slow, esp in FullDebug, hence we often run only a subset of tests (see the CI-setup)

Also what you can do is to check if you find tests that are related to what you modify. Oftentimes the test file names somehow reflect what is tested

Reply:
Oh Understood. 
Yes, generally I check for the local test (.cpp/.py file) related to what I have modified. But for the file above (_find_global_nodal_elemental_neighbours_process.cpp_ ), I didn't find anything. 
But, Anyways that general test  _kratos/python_scripts/run_tests.py_ I can run it as a safe side. 
Thanks :)

Reply:
here is one: https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/tests/cpp_tests/processes/test_find_nodal_neighbours_process.cpp

Reply:
Ohh I see. I haven't seen this :(
Inside this line 56, in turn will call the **execute()** of _find_global_nodal_elemental_neighbours_process.cpp_
Understood. Thanks :)

![image](https://user-images.githubusercontent.com/46954821/108381456-1d313100-7208-11eb-9281-6ba4de235b78.png)